src/GitHub.TeamFoundation.14/Connect/GitHubConnectSection.cs

```diff
@@ -182,7 +181,7 @@ public override void Initialize(IServiceProvider serviceProvider)
             // watch for new repos added to the local repo list
             var section = GetSection(TeamExplorerConnectionsSectionId);
             if (section != null)
-                sectionTracker = new SectionStateTracker(section, RefreshRepositories);
+                sectionTracker = new SectionStateTracker(section, () => RefreshRepositories(true));
```

Comment:
You can do the same `sectionIndex == 0` thing here too, there's only one `TeamExplorerConnectionsSectionId` section that will get tracked, and if it's updated it'll trigger for both of our sections.